csharp/src/Google.Protobuf/Collections/RepeatedField.cs

```diff
@@ -543,6 +543,20 @@ public override string ToString()
             }
         }
 
+#if NETSTANDARD2_0
+        /// <summary>
+        /// Creates a new span over the RepeatedField.  This is the most
+        /// efficient way to read and write values to the RepeatedField because
+        /// there are no internal null or bounds checks.  You are responsible
+        /// for ensuring no null values are written to the span.
+        /// </summary>
+        /// <returns>Span representation of the RepeatedField</returns>
+        public Span<T> AsSpan()

```

Comment:
Let's say I wanted to populate the repeated field with 1 million values. How do I resize the empty RepeatedField to the right size using the public API?

Reply:
I have another PR for that, but I will need to modify it slightly.  It preallocates the array but doesn't increase count, so, as you point out, it isn't going to work.  It will still be nice to have for reads.

Reply:
It would be good to first know what it the full API you're proposing. Isolated like this, it doesn't make much sense.
Btw, we just merged https://github.com/protocolbuffers/protobuf/pull/6317, so this PR might be out of date.